docs/deployments.md

```diff
@@ -0,0 +1,115 @@
+
+# Kubernetes Deployment Yönetimi
+
+Kubernetes'in Deployment objesi, belirlenen desired state'e uygun olarak bir veya birden fazla pod oluşturur ve bu state'i sürekli olarak mevcut durumla karşılaştırır. Bu işlem, uygulamanın istenilen durumda kalmasını sağlar ve gerektiğinde düzeltmeler yapar.
+
+## Deployment Nedir?
+
+Deployment, Kubernetes'teki bir uygulamanın birden fazla kopyasını yönetir. Bu obje, oluşturulacak pod'ların özelliklerini ve sayılarını belirtir ve bu sayede uygulamanın yüksek erişilebilirlik ve güvenilirlikle çalışmasını sağlar.
+
+## Temel Komutlar
+
+### Deployment Oluşturma
+
+```
+kubectl create deployment <isim> --image=<görüntü_ismi> --replicas=<kopya sayısı>
+```
+
+Örnek:
+```
+kubectl create deployment my_deployment --image=openjdk:21-ea-19-jdk-slim-buster --replicas=2
+```
+
+### Deployment Listeleme
+
+Deployment'ları listelemek için:
+```
+kubectl get deployments
+```
+
+Canlı olarak izlemek için `-w` parametresi ekleyebilirsiniz:
+```
+kubectl get deployments -w
+```
+
+### Pod Yönetimi
+
+Deployment içindeki pod'lardan birini silmek isterseniz:
+```
+kubectl delete pods deployment-pod-1
+```
+
+Bu durumda, eksik pod otomatik olarak yeniden oluşturulur.
+
+### Deployment Güncelleme
+
+Deployment'daki bir imajı değiştirmek için:
+```
+kubectl set image deployment/<deployment_name> <container_name>=<new_image>
+```
+
+Örnek:
+```
+kubectl set image deployment/my_deployment openjdk=21-ea-19-jdk-slim-buster=nginx
+```
+
+### Replika Sayısını Ayarlama
+
+Replika sayısını artırmak veya azaltmak için:
+```
+kubectl scale deployment <isim> --replicas=<yeni sayı>
+```
+
+### Deployment Silme
+
+Bir deployment'ı silmek için:
+```
+kubectl delete deployment <isim>
+```
+
+## Yaml ile Deployment Tanımlama
+
+Yaml dosyası kullanarak deployment tanımlayabilirsiniz. Örnek bir Yaml dosyası:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: simple-java-application
+  labels:
+    app: simple-java-application
+spec:
+  replicas: 2
+  selector:
```

Comment:
Selector kısmı için bir açıklama ekleyebilir misin? Nerelerde kullanılır neden bu alan tanımlanır vs vs gibi